### PR TITLE
Add support for BMK's "-upx" param

### LINKED ISSUE #49: Add support for new -upx build option
`bmk` now supports using [UPX](https://upx.github.io/) as a post `makeapp` step.
We should probably add a new toggle menu item for this. It would also be nice if it could detect the presence of the upx executable and therefore enable/disable the menu item appropriately.

### PR DESCRIPTION
**Pull request by GWRon**
- added enabler-menu entry
- added upx version information to "about box"
- (changed url from www.blitzmax.org to https://blitzmax.org)

Closes #49

`CanRunUPX()` could be replaced by a check of `GetUPX()`-return value. But this way we could extend the method to check for something more (capabilities etc).

### CODE REVIEW

**Comment by woollybah** [comment]
I did eventually get round to fixing the "www" alias on the website, although I do prefer urls without the needless `www.` prefix.

File: maxide.bmx [hunk 1]
```diff
@@ -87,7 +87,7 @@ Const EOL$="~n"
 
 Const ABOUT$=..
 "{bcc_version}.~n~n"+..
-"Please visit www.blitzmax.org for all your BlitzMax related needs!"
+"Please visit https://blitzmax.org for all your BlitzMax related needs!"
```
Comment (woollybah) [inline maxide.bmx]: I'm surprised the about text isn't localised too :-)

**Comment by GWRon** [comment]
Yes this would be a good thing - most people tend to prepend the "www." - to distinguish "organisations" ("Wellfare.Org") and "website" ("visit us at www.wellfare.org").
Should be a minor thing to get it running.

File: maxide.bmx (hunk 1 above)
Comment (GWRon) [reply maxide.bmx]: Can add a little commit to fix that ;-)

Comment (GWRon) [reply maxide.bmx]: :laughing: 
That line ... isn't even used in maxide.bmx ...

File: maxide.bmx [hunk 2]
```diff
@@ -7767,6 +7827,25 @@ Type TCodePlay
...
+		local platform:string = GetPlatform()
+		If platform = "emscripten" Or platform = "nx" Or platform = "ios" Then
+			Return False
+		End If
+		
+		Local upx:String = BlitzMaxPath() + "/bin/upx"
+		?win32
+		upx :+ ".exe"
+		?
+		If FileType(upx) = 1
```
Comment (woollybah) [inline maxide.bmx]: You shouldn't use seemingly random numbers when there are consts available. (e.g. FILETYPE_FILE)

When I change file types to an enum, all your code will break...

Comment (GWRon) [reply maxide.bmx]: Will change in the next commit.

File: maxide.bmx [hunk 3]
```diff
@@ -679,6 +685,18 @@ Type TAboutRequester Extends TRequester
 		Return GetProcessOutput(gppPath, "-dumpversion").Split("~n")[0]
 	EndMethod
 
+	Method GetUPX$()
+		Local upxPath:String = BlitzMaxPath() + "/bin/upx"
+		?Win32
+		upxPath :+ ".exe"
+		?
+		If FileType(upxPath) = 1
```
Comment (woollybah) [inline maxide.bmx]: You shouldn't use seemingly random numbers when there are consts available. (e.g. FILETYPE_FILE)

When I change file types to an enum, all your code will break...

File: maxide.bmx (hunk 1 above)
Comment (woollybah) [reply maxide.bmx]: Oh? Are you sure? We should just remove it if it's not used, because it's just taking up memory otherwise.

**Comment by GWRon** [comment]
Removed that about-string and made the subtitle in the about box localizable.

And ... replaced the FileType-check numbers with their constants.

**Pull request merged**